runtime/src/main/java/org/corfudb/runtime/view/MultiStreamView.java

```diff
@@ -80,161 +84,68 @@ public IStreamView copy(UUID source, UUID destination, long timestamp) {
         return new BackpointerStreamView(runtime, destination);
     }
 
-    /**
-     * Write an object to multiple streams, retuning the physical address it
-     * was written at.
-     * <p>
-     * Note: While the completion of this operation guarantees that the append
-     * has been persisted, it DOES NOT guarantee that the object has been
-     * written to the stream. For example, another client may have deleted
-     * the stream.
-     *
-     * @param object The object to append to the stream.
-     * @return The address this
+    /** Append to multiple streams simultaneously, possibly providing
+     * information on how to resolve conflicts.
+     * @param streamIDs     The streams to append to.
+     * @param object        The object to append to each stream.
+     * @param conflictInfo  Conflict information for the sequencer to check.
+     * @return              The address the entry was written to.
+     * @throws TransactionAbortedException      If the transaction was aborted by
+     *                                          the sequencer.
      */
-    public long write(Set<UUID> streamIDs, Object object) {
-        return acquireAndWrite(streamIDs, object, t -> true, t -> true);
-    }
+    public long append(@Nonnull Set<UUID> streamIDs, @Nonnull Object object,
+                       @Nullable TxResolutionInfo conflictInfo)
+        throws TransactionAbortedException {
+
+        // Go to the sequencer, grab an initial token.
+        TokenResponse tokenResponse = conflictInfo == null ?
+                // Token w/o conflict info
+                runtime.getSequencerView().nextToken(streamIDs, 1) :
```

Comment:
looks like we may want to make nextToken take a Nullable TxResolutionInfo parameter, to avoid these repeated if's on (conflictInfo == null). 

For now, perhaps just add a TODO comment